### PR TITLE
test: optimize dll test

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?
make ddl unit test running faster
### What is changed and how it works?
One loop in the hot path cost  almost 10s to execute, and several test cases depends on this function. The loop run 100 times to randomly select a range of 3 to check whether the result is right, this is unneccssary. We can reduce it to 20 times with check the lower/upper boundary and some random inner case is enough to make sure the logic is right.

Here is the result:
Before change:
```
>  go test -v -vet=off -p 10 -race github.com/pingcap/tidb/ddl
ok  	ok  	github.com/pingcap/tidb/ddl	200.443s
```

After change:
```
> go test -v -vet=off -p 10 -race github.com/pingcap/tidb/ddl
ok  	github.com/pingcap/tidb/ddl	118.154s
```

### Check List
Tests

- unit tests

Code changes

Side effects

Related changes

### CODE REVIEW

**Comment by glorv** [comment]
@zimulala  PTAL

File: ddl/db_test.go [hunk 1]
```diff
@@ -1029,9 +1029,13 @@ LOOP:
 		return
 	}
 
-	// test index range
-	for i := 0; i < 100; i++ {
-		index := rand.Intn(len(keys) - 3)
+	// test index range, test lower/upper boundary and random inner cases
```
Comment (zimulala) [inline ddl/db_test.go]:
```suggestion
	// Test the index range. Test lower/upper boundary and random inner cases.
```

File: ddl/db_test.go [hunk 2]
```diff
@@ -1029,9 +1029,13 @@ LOOP:
...
+	step := len(keys) / 20
+	for i := 0; i <= 20; i++ {
+		index := i * step
+		if index >= len(keys)-3 {
```
Comment (zimulala) [inline ddl/db_test.go]:
```suggestion
		if index > len(keys)-3 {
```

**Pull request merged**